script.js

```diff
@@ -17,3 +17,210 @@ for (let i = 0; i < mobileItems.length; i += 1) {
+                  <a href="#">${element.technologies[1]}</a>
+              </li>
+              <li>
+                  <a href="#">${element.technologies[2]}</a>
+              </li>
+          </ul>
+      </div>
+      <button type="submit" class="card-btn" id="card${element.id}">See Project</button>
+  </div>
+</div>`;
+}
+
+// popup window
+
+function getPopupId(id) {
+  for (let j = 0; j < projectsArr.length; j += 1) {
+    if (projectsArr[j].id === id) {
+      const popupWindow = document.getElementById('popup');
+      popupWindow.innerHTML += `<div class="popup-content">
+          <div class="popup-header">
+              <h2>${projectsArr[j].name}</h2>
+              <button class="cancle" id="cancle${projectsArr[j].id}"><img src="./images/Icon - Cancel.png" alt="cancle icon"></button>
+          </div>
+          <ul>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[0]}</a>
+              </li>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[1]}</a>
+              </li>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[2]}</a>
+              </li>
+          </ul>
+          <div class="popup-footer">
+              <img class="popup-img" src="${projectsArr[j].featuredImg}" alt="">
+              <div class="footer-content">
+                  <p>${projectsArr[j].desc}</p>
+                  <div class="popup-btn">
+                      <a href="${projectsArr[j].liveLink}">See live <img src="./images/live.png" alt=""></a>
+                      <a href="${projectsArr[j].sourceLink}">See source <img src="./images/Icon -GitHub.png" alt=""></a>
+                  </div>
+              </div>
+          
+          </div>
+          </div>`;
+    }
+  }
+  const canclex = document.querySelector(`#cancle${id}`);
+  canclex.addEventListener('click', () => {
+    const popup = document.querySelector('.popup');
+    popup.style.display = 'none';
+  });
+}
+
+const primarybtn = document.querySelector('.primary-btn');
+primarybtn.addEventListener('click', () => {
+  const popup = document.querySelector('.popup');
+  popup.style.display = 'block';
+  popup.innerHTML = '';
+  getPopupId(workproject.id);
+});
+
+for (let i = 0; i < projectsArr.length; i += 1) {
+  const cardBtn = document.querySelector(`#card${projectsArr[i].id}`);
+  cardBtn.addEventListener('click', () => {
+    const popup = document.querySelector('.popup');
+    popup.style.display = 'block';
+    popup.innerHTML = '';
+    getPopupId(projectsArr[i].id);
+  });
+}
+
+// form validation
+
+const form = document.getElementById('contact-form');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const email = document.getElementById('email').value;
+
+  const regex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/;
+
+  const formErrorMsg = document.querySelector('.form-error-msg');
+  if (!regex.test(email)) {
+    formErrorMsg.innerHTML = 'error';
+  } else {
```

Comment:
- [ ] Why don't you say something to users describing the error that has occurred rather than just saying `error` to them? :smile:

Reply:
my teammates made changes to the error message but I made mistake committing in my project and this is why somehow I bring back my previous code..I don't know why it was happed